histomicstk/cli/NucleiDetection/NucleiDetection.py

```diff
@@ -17,20 +17,58 @@
 logging.basicConfig(level=logging.CRITICAL)
 
 
+def read_input_image(args, process_whole_image=False):
+    # read input image and check if it is WSI
+    print('\n>> Reading input image ... \n')
+
+    ts = large_image.getTileSource(args.inputImageFile, style=args.style)
+
+    ts_metadata = ts.getMetadata()
+
+    print(json.dumps(ts_metadata, indent=2))
+
+    is_wsi = ts_metadata['magnification'] is not None
+
+    return ts, is_wsi
+
+
+def image_inversion_flag_setter(args=None):
+    # generates image inversion flags
+    invert_image, default_img_inversion = False, False
+    if args.ImageInversionForm == "Yes":
+        invert_image = True
+    if args.ImageInversionForm == "No":
+        invert_image = False
+    if args.ImageInversionForm == "default":
+        default_img_inversion = True
+    return invert_image, default_img_inversion
+
+
 def detect_tile_nuclei(slide_path, tile_position, args, it_kwargs,
-                       src_mu_lab=None, src_sigma_lab=None):
+                       src_mu_lab=None, src_sigma_lab=None, invert_image=False,
+                       style=None, default_img_inversion=False):
+
+    # Flags
+    single_channel = False
 
     # get slide tile source
-    ts = large_image.getTileSource(slide_path)
+    ts = large_image.getTileSource(slide_path, style=style)
 
     # get requested tile
     tile_info = ts.getSingleTile(
         tile_position=tile_position,
         format=large_image.tilesource.TILE_FORMAT_NUMPY,
         **it_kwargs)
 
-    # get tile image
-    im_tile = tile_info['tile'][:, :, :3]
+    # get tile image & check number of channels
+    print('The given shape of image is ', tile_info['tile'].shape)
```

Comment:
I'd rather remove this print statement; it makes the job record harder to look through.